packages/gatsby-transformer-video/src/ffmpeg.js

```diff
@@ -0,0 +1,492 @@
+import { resolve, parse } from "path"
+import { performance } from "perf_hooks"
+
+import { createContentDigest } from "gatsby-core-utils"
+import { createRemoteFileNode } from "gatsby-source-filesystem"
+import { pathExists, stat, copy, writeFile } from "fs-extra"
+import ffmpeg from "fluent-ffmpeg"
+import fg from "fast-glob"
+import imagemin from "imagemin"
+import imageminGiflossy from "imagemin-giflossy"
+import imageminMozjpeg from "imagemin-mozjpeg"
+import PQueue from "p-queue"
+import sharp from "sharp"
+
+import profileH264 from "./profiles/h264"
+import profileH265 from "./profiles/h265"
+import profileVP9 from "./profiles/vp9"
+import profileWebP from "./profiles/webp"
+import profileGif from "./profiles/gif"
+
+export default class FFMPEG {
+  constructor({ rootDir, cacheDir, ffmpegPath, ffprobePath, profiles }) {
+    this.queue = new PQueue({ concurrency: 1 })
+    this.cacheDir = cacheDir
+    this.rootDir = rootDir
+    this.profiles = profiles
+    this.totalVideos = 0
+    this.currentVideo = 0
+
+    if (ffmpegPath) {
+      ffmpeg.setFfmpegPath(ffmpegPath)
+    }
+    if (ffprobePath) {
+      ffmpeg.setFfprobePath(ffprobePath)
+    }
+  }
+
+  // Execute FFPROBE and return metadata
+  executeFfprobe = path =>
+    new Promise((resolve, reject) => {
+      ffmpeg(path).ffprobe((err, data) => {
+        if (err) reject(err)
+        resolve(data)
+      })
+    })
+
+  // Execute FFMMPEG and log progress
+  executeFfmpeg = async ({ ffmpegSession, cachePath, loggingPrefix }) => {
+    let startTime
+    let lastLoggedPercent = 0.1
+
+    return new Promise((resolve, reject) => {
+      ffmpegSession
+        .on(`start`, commandLine => {
+          console.log(`${loggingPrefix} Executing:\n\n${commandLine}\n`)
+          startTime = performance.now()
+        })
+        .on(`progress`, progress => {
+          if (progress.percent > lastLoggedPercent + 10) {
+            const percent = Math.floor(progress.percent)
+            const elapsedTime = Math.ceil(
+              (performance.now() - startTime) / 1000
+            )
+            const estTotalTime = (100 / percent) * elapsedTime
+            const estTimeLeft = Math.ceil(estTotalTime - elapsedTime)
+            const loggedTimeLeft =
+              estTimeLeft !== Infinity && ` (~${estTimeLeft}s)`
+
+            console.log(`${loggingPrefix} ${percent}%${loggedTimeLeft}`)
```

Comment:
not in the scope of this PR but as a follow up PR: other plugins are using ProgressBar for the logging ... [Code Search ProgressBar](https://github.com/gatsbyjs/gatsby/search?q=progressbar&unscoped_q=progressbar)

Reply:
time calculation can be done then by the ProgressBar:

https://github.com/gatsbyjs/gatsby/blob/db36d6995659c190b4aefda2d60828d48a430611/packages/gatsby-plugin-sharp/src/utils.js#L12-L19

see Tokens: https://www.npmjs.com/package/progress#tokens

Reply:
+1 :)